core/lib/Thelia/Core/Template/TemplateHelper.php

```diff
@@ -12,321 +12,94 @@
 
 namespace Thelia\Core\Template;
 
-use Symfony\Component\Filesystem\Filesystem;
-use Thelia\Model\ConfigQuery;
-use Thelia\Log\Tlog;
-use Thelia\Core\Translation\Translator;
-
-class TemplateHelper
+/**
+ * Class TemplateHelper
+ * @author Franck Allimant <franck@cqfdev.fr>
+ * @package Thelia\Core\Template
+ *
+ * @deprecated use TheliaTemplateHelper service (thelia.template_helper) instead. This class will be DELETED in 2.3
+ */
+class TemplateHelper implements TemplateHelperInterface
 {
-    const WALK_MODE_PHP = 'php';
-    const WALK_MODE_TEMPLATE = 'tpl';
-
     /**
      * This is a singleton
-
      */
     private static $instance = null;
 
-    private function __construct()
+    private function __construct(TemplateHelperInterface $templateHelper)
     {
+        $this->templateHelper = $templateHelper;
     }
 
+    /**
+     * @deprecated use TheliaTemplateHelper service (thelia.template_helper) instead.
+     */
     public static function getInstance()
     {
         if (self::$instance == null) {
-            self::$instance = new TemplateHelper();
+            throw new \LogicException("This class should be initialized before getInstance() call.");
```

Comment:
The singleton is supposed to be initialized by TheliaTemplateHelper when it is instanciated at bootstrap. 

Thus this exception is not thrown in normal bootstrap process. I've coded it "just in case".